packages/babel-plugin-transform-decorators-2/test/fixtures/examples/logger/exec.js

```diff
@@ -0,0 +1,31 @@
+const calls = [];
+
+function log(message) {
+  return function (descriptor) {
+    let oldFunc = descriptor.value;
+    descriptor.value = function(...args) {
```

Comment:
Actually, we'll probably be passing the element descriptor to the decorator and not the property descriptor. I just had a doubt on how that'd work when you change the key of the descriptor you've returned and that's why I made the call to implement it like this. Once I get a little clarification on assumption #4 from @littledan or someone, I'll make the desired changes